qp_klp/Step.py

```diff
@@ -326,23 +328,187 @@ def _generate_prep_file(self, config, input_file_path, seqpro_path,
 
         return gpf_job
 
-    def _generate_commands(self):
-        out_dir = self.pipeline.output_path
-        qclient = self.qclient
+    def _helper_process_fastp_report_dirs(self):
+        report_dirs = []
+
+        for root, dirs, files in walk(self.pipeline.output_path):
+            for dir_name in dirs:
+                if dir_name == 'fastp_reports_dir':
+                    # generate the full path for this directory before
+                    # truncating everything up to the QCJob directory.
+                    full_path = join(root, dir_name).split('QCJob/')
+                    report_dirs.append(join('QCJob', full_path[1]))
+
+        if report_dirs:
+            report_dirs.sort()
+            return 'tar zcvf reports-QCJob.tgz ' + ' '.join(report_dirs)
+        else:
+            # It is okay to return an empty list of commands if reports_dirs
+            # is empty. Some pipelines do not generate fastp reports.
+            return []
+
+    def _helper_process_blanks(self):
+        results = [x for x in listdir(self.pipeline.output_path) if
+                   x.endswith('_blanks.tsv')]
+
+        if len(results) > 0:
+            return 'tar zcvf sample-files.tgz' + ' ' + ' '.join(results)
+
+    def _helper_process_blank_files(self):
+        blanks_files = [a_file for a_file in listdir(self.pipeline.output_path)
+                        if a_file.endswith('_blanks.tsv')]
+
+        # where proj[2] is a Qiita ID and proj[1] is a project uploads path.
+        uploads_folders = {proj[2]: proj[1] for proj in self.special_map}
+
+        cmds = []
+        for blanks_file in blanks_files:
+            # assume filename like:
+            # 180716_M05314_0106_000000000-BYB7J_Metcalf_NIJ_14989_blanks.tsv
+            qiita_id = blanks_file.split('_')[-2]
+            uploads_folder = uploads_folders[qiita_id]
+
+            cmds.append(f"mv {blanks_file} {uploads_folder}")
+
+        return cmds
+
+    def _helper_process_fastq_files(self):
+        cmds = []
+
+        valid_dir_names = [join(self.pipeline.output_path, 'QCJob', a_dir) for
+                           a_dir in ['amplicon', 'filtered_sequences',
+                                     'trimmed_sequences']]
+
+        data_dirs = [a_dir for a_dir in valid_dir_names if
+                     exists(a_dir) and isdir(a_dir)]
+
+        uploads_folders = {proj[2]: proj[1] for proj in self.special_map}
```

Comment:
... if no need to upload files, do we need this?

Reply:
I believe so. We still use this code to help move the filtered fastq files to the uploads directory:
https://github.com/charles-cowart/qp-knight-lab-processing/blob/535cf6b28f6194051aade2e1b34c4cceb17bf9be/qp_klp/Step.py#L399

Reply:
We shouldn't, please remove